opal/mca/if/linux_ipv6/if_linux_ipv6.c

```diff
@@ -126,8 +127,11 @@ static int if_linux_ipv6_open(void)
             opal_if_t *intf;
 
             if (!hexdecode(addrhex, a6.s6_addr, sizeof a6.s6_addr)) {
-                opal_output(0, LOG_PREFIX "skipped interface %s: failed to hexdecode %s\n",
-                            ifname, addrhex);
+                char hostname[OPAL_MAXHOSTNAMELEN + 1];
```

Comment:
`OPAL_MAXHOSTNAMELEN` already accounts for `+ 1`: https://github.com/open-mpi/ompi/blob/v4.0.1/opal/include/opal_config_bottom.h#L295-L302

Reply:
Please use:
```
char hostname[OPAL_MAXHOSTNAMELEN];
gethostname(hostname, sizeof hostname);
```
(as in the rest of ompi codebase), because it lets `gethostname` write the final `\0` when the hostname reaches `MAXHOSTNAMELEN` (which is `OPAL_MAXHOSTNAMELEN - 1`).

Reply:
Good catch; thanks.

I just checked the man page for `gethostname(2)` on RHEL 6 and RHEL 7: it looks like `gethostname()` does *not* guarantee to copy over the null terminator if `len` is too small. 😲

As you cite, however, this is likely a larger issue for the rest of the code base, and likely mitigated by having `OPAL_MAXHOSTNAMELEN` be (the_max+1) (even though glibc `gethostname(2)` does not state that there is any relationship between `MAXHOSTNAMELEN` or `HOST_NAME_MAX` and the maximum string length returned by `gethostname(3)`  ...but glibc `gethostname(3p)` *does* state that `HOST_NAME_MAX` is the max length.  🤷‍♂  FWIW, MacOS 10.14 `gethostname(3)` states that the max hostname length will be no less than `_POSIX_HOST_NAME_MAX`, which is effectively a synonym for `MAXHOSTNAMELEN`).

Probably the simplest "99%" fix to do across the code base is therefore:

```c
char hostname[OPAL_MAXHOSTNAMELEN] = {0};
gethostname(hostname, sizeof(hostname));
```

If we assume that `OPAL_MAXHOSTNAMELEN` is, indeed, 1 greater than the max, then the `={0}` will be enough to guarantee that the result is null terminated.

Regardless, this seems like a larger scope (but still probably trivial) issue than this PR.  I put the `={0}` compromise in this PR; I'll open a "help wanted" github issue to apply this kind of trivial fix elsewhere in the code base.

Reply:
```
char hostname[OPAL_MAXHOSTNAMELEN] = {0};
gethostname(hostname, sizeof(hostname));
```
The first line has no effect. If `len(gethostname()) < sizeof(hostname)`, then `gethostname` will write `\0` by itself, otherwise the second line will overwrite the final `\0`.

Reply:
Blarg.  That should have been `sizeof(hostname)-1`, which would have protected against the case where the hostname > `OPAL_MAXHOSTNAMELEN`.

@hppritcha unfortunately merged this PR already, so I can't fix it on this PR.  It's *almost* not worth another PR... but I'll be annoyed with Past Jeff someday if I come across this code again in the future and wonder why the `{0}` is there without the corresponding `-1`.  Arrgh.